src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests_Dialog.cs

```diff
@@ -2006,114 +2006,6 @@ public interface DDD
 assertGenerateTypeDialogOptions: new GenerateTypeDialogOptions(true, TypeKindOptions.Interface, false));
         }
 
-        [WorkItem(861362)]
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
-        public void GenerateTypeInMemberAccessExpression()
-        {
-            TestWithMockedGenerateTypeDialog(
-initial: @"class Program
-{
-    static void Main(string[] args)
-    {
-        var s = [|$$A.B|];
-    }
-}",
-languageName: LanguageNames.CSharp,
-typeName: "A",
-expected: @"class Program
-{
-    static void Main(string[] args)
-    {
-        var s = A.B;
-    }
-}
-
-public class A
-{
-}",
-accessibility: Accessibility.Public,
```

Comment:
@CyrusNajmabadi and I chatted about this, and I think we agree that we still want to handle member-access expressions. However, we should process them by attempting to generate a type for the name to the left of the name. So, we should still start from the right, but take the name to the left.

**Example 1**:

```C#
class C
{
    void M()
    {
        var x = A.B;
    }
}
```

Should generate a class called "A".

**Example 2**:

```C#
class C
{
    void M()
    {
        var x = A.B.C;
    }
}
```

Should generate a class called "B" in a namespace called "A".

Does that seem reasonable to you?